// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

message ComputeFirewallPolicy {
  string name = 1;
  string description = 2;
  string fingerprint = 3;
  string self_link = 4;
  string self_link_with_id = 5;
  int64 rule_tuple_count = 6;
  string display_name = 7;
  string parent = 8;
}

message ApplyComputeFirewallPolicyRequest {
  ComputeFirewallPolicy resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteComputeFirewallPolicyRequest {
  string service_account_file = 1;
  ComputeFirewallPolicy resource = 2;
}

message ListComputeFirewallPolicyRequest {
  string service_account_file = 1;
  string Parent = 2;
}

message ListComputeFirewallPolicyResponse {
  repeated ComputeFirewallPolicy items = 1;
}

service ComputeFirewallPolicyService {
  rpc ApplyComputeFirewallPolicy(ApplyComputeFirewallPolicyRequest) returns (ComputeFirewallPolicy);
  rpc DeleteComputeFirewallPolicy(DeleteComputeFirewallPolicyRequest) returns (google.protobuf.Empty);
  rpc ListComputeFirewallPolicy(ListComputeFirewallPolicyRequest) returns (ListComputeFirewallPolicyResponse);
}
